src/components/Modal/index.tsx

```diff
@@ -93,6 +99,19 @@ export default function Modal({
         <div className="modal__content">{derivedChildren}</div>
         {buttons.length > 0 && (
           <div className="modal__footer">
+            {checkboxContent && (
+              <div className="checkbox">
+                <div
+                  onClick={(): void => {
+                    if (setCheckbox !== undefined) {
+                      setCheckbox(!checkbox);
+                    }
+                  }}
+                  style={checkbox ? { backgroundColor: '#C56E5B' } : {}}
+                />
+                <p>{checkboxContent}</p>
+              </div>
+            )}
             {buttons.map((props, i) => (
               <ModalButton {...props} key={i} />
             ))}
```

Comment:
On mobile the button should be right aligned and the checkbox should be left aligned (although make sure this doesn't mess up the other modals where the button should be middle aligned where there is no checkbox)